src/traversing.js

```diff
@@ -168,10 +159,18 @@ jQuery.each({
 		return elem.contentDocument || jQuery.merge( [], elem.childNodes );
 	}
 }, function( name, fn ) {
+	var // Sorting (which implicitly de-dupes) is required for non-adjacent traversals
+		sort = name !== "prev" && name !== "next",
```

Comment:
Also, things may have changed in 4 years since you reported [trac-7964](http://bugs.jquery.com/ticket/7964). Do you recall if that ticket was based on some real-life perf issue? At this point we've reduced the optimization to just those two methods so it might be worth simplifying the code and seeing who complains.